src/raft.c

```diff
@@ -575,6 +575,19 @@ static void handleEndClientSession(RedisRaftCtx *rr, raft_entry_t *entry)
     }
 }
 
+static void clearClientSessions(RedisRaftCtx *rr)
+{
+    void *client_session;
+    RedisModuleDictIter *iter = RedisModule_DictIteratorStartC(rr->client_session_dict, "^", NULL, 0);
+
+    while (RedisModule_DictNextC(iter, NULL, (void **) &client_session) != NULL) {
+        freeClientSession(client_session);
+    }
+    RedisModule_DictIteratorStop(iter);
+    RedisModule_FreeDict(rr->ctx, rr->client_session_dict);
+    rr->client_session_dict = RedisModule_CreateDict(rr->ctx);
```

Comment:
We don't have to free and alloc the whole dictionary here, but we should make sure that we free it at `RedisRaftCtxClear()` (+ please make sure that we actually free there the inner objs memory, for other dicts as well).

Reply:
So, I wasn't sure if we could delete items while iterating without breaking the iteration.

And, I wasn't sure if its actually less performance to do many individual deletes from the dict, vs just deleting and recreating the dict.

that's why I went this way, if we think its better in other way, that's ok too.

Reply:
k, checked now on Redis - indeed it seems like your practice is the right one, so let's keep it.